code/datums/components/submerge_mod.dm

```diff
@@ -0,0 +1,55 @@
+//Allows an atom to contribute to the submerging effects of a turf
+//Submerge height is additive, so best to avoid spamming multiple atoms on a turf with this component/very high values unless you want wack results.
+/datum/component/submerge_modifier
+	///Height in pixels a target is submerged by the parent of this component
+	var/submerge_height
+
+/datum/component/submerge_modifier/Initialize(_submerge_height = 10)
+	if(!ismovable(parent))
+		return COMPONENT_INCOMPATIBLE
+	submerge_height = _submerge_height
+
+/datum/component/submerge_modifier/RegisterWithParent()
+	var/atom/movable/owner = parent
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(move_submerge_element))
+	for(var/atom/movable/AM AS in owner.loc) //we remove the submerge and reapply after owner is connect
+		AM.set_submerge_level(null, owner.loc, duration = 0.1)
```

Comment:
Otherwise it will permanently screw up offsets for anything submerged on the tile. Pretty much a lot of this code is to avoid an atom level var, really.